extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/internal/dmxoverethernet/IpNode.java

```diff
@@ -0,0 +1,171 @@
+        }
+    }
+
+    /**
+     * constructor with address and port
+     *
+     * @param addrString
+     *            domain name or IP address as string representation
+     * @param port
+     *            UDP port of the receiver node
+     */
+    public IpNode(String addrString, int port) {
+        setPort(port);
+        setInetAddress(addrString);
+    }
+
+    /**
+     * sets the node address
+     *
+     * @param addrString
+     *            domain name or IP address as string representation
+     */
+    public void setInetAddress(String addrString) {
+        try {
+            this.address = InetAddress.getByName(addrString);
+        } catch (UnknownHostException e) {
+            this.address = null;
+            logger.warn("could not set address from {}", addrString);
+        }
+    }
+
+    /**
+     * set the node address
+     *
+     * @param address
+     *            inet address
+     */
+    public void setInetAddress(InetAddress address) {
+        this.address = address;
+    }
+
+    /**
+     * sets the node port
+     *
+     * @param port
+     *            UDP port of the receiver node
+     */
+    public void setPort(int port) {
+        this.port = port;
+    }
+
+    /**
+     * get this nodes port
+     *
+     * @return
+     *         UDP port
+     */
+    public int getPort() {
+        return this.port;
+    }
+
+    /**
+     * get this nodes address
+     *
+     * @return
+     *         address as InetAddress
+     */
+    public InetAddress getAddress() {
+        return address;
+    }
+
+    public String getAddressString() {
+        String addrString = address.getHostAddress();
+        return addrString;
+    }
+
+    /**
+     * convert node to String
+     *
+     * @return
+     *         string representation of this node (address:port)
+     */
+    @Override
+    public String toString() {
+        if (this.address == null) {
+            return "(null):" + String.valueOf(this.port);
+        }
+        return this.address.toString() + ":" + String.valueOf(this.port);
+    }
+
+    /**
+     * create list of nodes from string
+     *
+     * @param addrString
+     *            input string, format: address1[:port],address2
+     * @param defaultPort
+     *            default port if none is given in the string
+     * @return
+     *         a List of IpNodes
+     */
+    public static List<IpNode> fromString(String addrString, int defaultPort) {
+        List<IpNode> ipNodes = new ArrayList<IpNode>();
+        int port;
+
+        for (String singleAddrString : addrString.split(",")) {
+            Matcher addrMatch = ADDR_PATTERN.matcher(singleAddrString);
+            if (addrMatch.matches()) {
+                port = (addrMatch.group(2).isEmpty()) ? defaultPort : Integer.valueOf(addrMatch.group(2));
+                ipNodes.add(new IpNode(addrMatch.group(1), port));
+            } else {
+                logger.warn("Node definition {} is not valid.", singleAddrString);
```

Comment:
I would say that throwing IllegalArgumentExceptions are the best solution to this. They will bubble up in the stack and you can catch them higher in your logic where you have a chance to set the Thing status (or, if the Thing can actually stay ONLINE, because most of it still works, you can then log a warning to the user that he misconfigured something - but at that place, it can be done in the context of a specific thing, so that will help the user to identify the problem).